### PR TITLE
Set -Wredundant-constraints and fix the errors

### PR DESCRIPTION
**Pull request by lepsa**
Enabling the redundant constraint warning and cleaning up the resulting errors.

## Checklist

 - [:heavy_check_mark:] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [:heavy_check_mark:] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -110,5 +110,8 @@ result-*
 # emacs misc
 .dir-locals.el
 
+# vscode config
+.vscode
```
Comment (elland) [inline .gitignore:114]: Maybe have that in your local `~/.gitignore`? 👍

Comment (lepsa) [reply .gitignore]: I've removed it from the PR.

**Review by elland — APPROVED** [review]
Looks fine to me 👯, but CI seems unsure.

File: services/brig/src/Brig/IO/Intra.hs [hunk 2]
```diff
@@ -508,14 +502,14 @@ notifyContacts events orig route conn = do
     contacts = lookupContactList orig
 
     teamContacts :: m [UserId]
-    teamContacts = screenMemberList =<< getTeamContacts orig
+    teamContacts = screenMemberList <$> getTeamContacts orig
```
Comment (elland) [inline services/brig/src/Brig/IO/Intra.hs:505]: Sneaky refactor? 😄

File: services/gundeck/src/Gundeck/Push/Native/Serialise.hs [hunk 3]
```diff
@@ -32,11 +35,11 @@ import Gundeck.Push.Native.Types
 import Gundeck.Types
 import Imports
 
-serialise :: HasCallStack => NativePush -> UserId -> Transport -> IO (Either Failure LT.Text)
+serialise :: HasCallStack => NativePush -> UserId -> Transport -> Either Failure LT.Text
```
Comment (elland) [inline services/gundeck/src/Gundeck/Push/Native/Serialise.hs:38]: 👏

**Pull request merged**